### DISCUSSION TITLE
Wrong RegExp in Series overrides cause blank graph

### DISCUSSION THREAD

**Discussion by b1zzu** [grafana/grafana#36996] [Dashboards, Panels & Visualizations]
Grafana: 4.6.3
Datasource: Influxdb 
Host OS: CentOS 7

Hi,

I have created a new graph with a single metric to which I have given the alias **/etc/hostname**. Then I want in **Display** -> **Series overrides** and I have overwritten the color of my metrics.
![grafanaaliasregexerror](https://user-images.githubusercontent.com/7964685/39421448-5f858618-4c69-11e8-9ce8-c4b408ecf6d4.png)

After that, I have saved the dashboard, and I have reloaded the page, and the graph was blank.
![grafanaaliaserrorresult](https://user-images.githubusercontent.com/7964685/39421503-a50848c4-4c69-11e8-940e-1b5c348d3b9a.png)

In the Javascript console I have this error:
```
vendor.556d03624dd5cdf1009d.js:43 TypeError: Cannot read property '1' of null
    at Object.a.stringToJsRegex (app.5331f559bd9a1bed9a93.js:1)
    at r (0.be20b78823b4c9d93a84.js:1)
    at t.applySeriesOverrides (0.be20b78823b4c9d93a84.js:1)
    at e.onRender (0.be20b78823b4c9d93a84.js:7)
    at o.emit (vendor.556d03624dd5cdf1009d.js:15)
    at t.emit (app.5331f559bd9a1bed9a93.js:1)
    at e.t.render (0.be20b78823b4c9d93a84.js:1)
    at 0.be20b78823b4c9d93a84.js:1
    at f.$broadcast (vendor.556d03624dd5cdf1009d.js:43)
    at 0.be20b78823b4c9d93a84.js:1
```

I fixed the problem by changing **/etc/hostname** in **/^\/etc\/hostname$/** in **Series overrides** becouse otherwise it was detected as a broken RegEx.

I think in this case Grafana should give a more useful error, for example by detecting immediately that the RegEx was wrong, and by giving a visual error wit explained the reason for the problem. This because I have invested more than four hours before finding the problem, and maybe other users have, or could have the same problem in the future.

Bye,
Davide

**Comment by jschill** [comment]
**Problem:** Add a _"series specific override"_ with a leading slash (see screenshots from @b1zzu). When you blur the input field you will get a javascript error, since all overrides starting with a slash is treated as a regex.

**Possible solution:** Like @b1zzu mention we need to provide the user some feedback when the regex is invalid. For these kind of errors we could use our event emitter and the "**alert-error**" event.

This is how the emitter is used:
1. first import appEvents.
`import { appEvents } from 'app/core/core';`
2. then emit the event in case of error (note that the message is inside of an array):
`appEvents.emit('alert-error', ['RegEx error', 'Remove the leading slash if it shouldn\'t be treated as RegEx']);`

TODO:
- [ ] Check if the RegEx is valid. If not, trigger the alert-error event. (https://github.com/grafana/grafana/blob/master/public/app/core/utils/kbn.ts#L236)
- [ ] Add test to https://github.com/grafana/grafana/blob/master/public/app/core/specs/kbn.test.ts

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
